### PR TITLE
persist: prep for upcoming changes to data blob encoding format

### PR DESCRIPTION
**Pull request by danhhz**
See individual commits for details

### CODE REVIEW

File: src/persist/src/indexed/arrangement.rs [hunk 1]
```diff
@@ -973,24 +973,30 @@ mod tests {
     }
 
     // Keys are randomly generated, so clear them before we do any comparisons.
-    fn cleared_unsealed_keys(batches: &[UnsealedBatchMeta]) -> Vec<UnsealedBatchMeta> {
+    // Also clear size_bytes so the tests don't have to be updated every time
```
Comment (ruchirK) [inline src/persist/src/indexed/arrangement.rs:976]: love this

File: src/persist/src/indexed/encoding.rs [hunk 2]
```diff
@@ -1537,4 +1538,40 @@ mod tests {
...
+            trace.encode(&mut trace_buf);
+            (unsealed_buf.len(), trace_buf.len())
+        }
+
+        let record_size_bytes = DataGenerator::default().record_size_bytes;
+        // Print all the sizes into one assert so we only have to update one
+        // place if sizes change.
+        assert_eq!(
+            format!(
+                "1={:?} 25={:?} 1000={:?} 1000/100={:?}",
```
Comment (ruchirK) [inline src/persist/src/indexed/encoding.rs:1568]: two small nits:

1. maybe it makes sense to also have one check in the millions of records here? I'm kind of undecided
2. the 1000/100 is meant to indicate "1000 records in 100 batches" right? I wonder if maybe that makes more sense with 1/1=... 25/1=... 1000/1... and then 1000/100. 

Feel free to take or leave both of these

**Review by ruchirK — APPROVED** [review]
LGTM!

File: src/persist/src/indexed/arrangement.rs (hunk 1 above)
Comment (danhhz) [reply src/persist/src/indexed/arrangement.rs:976]: Yeah, I saw this one coming :)

https://github.com/MaterializeInc/materialize/blob/8cdf118ddd8de7825c4f665cf43f7c8ce611f3b3/src/persist/src/indexed/arrangement.rs#L1356

File: src/persist/src/indexed/encoding.rs (hunk 2 above)
Comment (danhhz) [reply src/persist/src/indexed/encoding.rs]: > maybe it makes sense to also have one check in the millions of records here? I'm kind of undecided

I gave it a shot but looks like the test takes way too long. Seems to be pretty close to 1000x the 1000 case, anyway.

> the 1000/100 is meant to indicate "1000 records in 100 batches" right? I wonder if maybe that makes more sense with 1/1=... 25/1=... 1000/1... and then 1000/100.

oops, it was supposed to be the raw params (records/records per batch) but I originally wrote the test to be 100, then later changed it to 10 and forgot to update this format string. your suggestion is better anyway, updated

**Review by danhhz — COMMENTED** [review]
TFTR!

**Pull request merged**